### PR TITLE
Setup updated + cm11/cm12 support + serial reviewed

### PR DESCRIPTION
**Pull request by rferrazz**
Setup script updated to use distutils
Reviewed serial to use more modern and simplier methods for transmission
Added cm11/cm12 support (tested)

### CODE REVIEW

File: x10/controllers/abstract.py [hunk 1]
```diff
@@ -103,31 +107,15 @@ def _set_off(self):
         self._handle.setRTS(False)
 
     def write(self, aSequence):
-        logger.debug("Writing %s", ["0x%02x" % i for i in aSequence])
-
-        byte_count = len(hex(aSequence)) - 2
-
-        mask = 1 << (byte_count - 1)
+        print (">", hex(aSequence))
```
Comment (glibersat) [inline x10/controllers/abstract.py]: could you use logging or remove this statement please?

**Comment by glibersat** [comment]
Thanks for this patch, this is awesome! :-)

File: x10/controllers/abstract.py [hunk 2]
```diff
@@ -103,31 +107,15 @@ def _set_off(self):
...
-            time.sleep(self.DELAY_BIT)
-            
-            # Move to next bit in sequence
-            aSequence = aSequence << 1
-
+    def read(self):
+        res = self._handle.read()
+        if res == "":
+            return res
+        print("<", hex(ord(res)))
```
Comment (glibersat) [inline x10/controllers/abstract.py]: same here : logging or please remove

**Comment by rferrazz** [comment]
updated!

**Comment by glibersat** [comment]
Thanks a lot Riccardo for your work, everything's merged.

**Pull request merged**